fixes test/unit/jobs/test_job_output_checker.py; strange default?

caleb-easterly opened the pull request:
Unclear on the implications of this. 

In `lib/galaxy/tools/parser/interface.py`, the default for `ToolStdioRegex.error_level` was `"fatal"`. The following test passes under Python2 but fails under Python3: 

https://github.com/galaxyproject/galaxy/blob/01ba96a7acda5f321e3f3f0975909615122a7880/test/unit/jobs/test_job_output_checker.py#L62-L68

This is because, in the following

https://github.com/galaxyproject/galaxy/blob/01ba96a7acda5f321e3f3f0975909615122a7880/lib/galaxy/jobs/error_level.py#L21-L26

`'fatal' > 0 and 'fatal' <= StdioErrorLevel.MAX` is `false` in Python 2 but throws an error in Python 3. It seems like these shouldn't be compared at all...but I might not have fixed it in the right way.

xref. #1715

caleb-easterly committed:
fixes test/unit/jobs/test_job_output_checker.py

martenson commented:
@galaxybot test this

mvdbeek reviewed (approved):
Thanks @caleb-easterly!

dannon commented:
This is the correct fix, but we should probably use the constant `StdioErrorLevel.FATAL` (which is 3).  Want to push one more changeset, or should I?  (and you can drop the TODO as well, since it's suggesting to use the constant here)

caleb-easterly commented:
I probably wouldn't get to it for an hour or so - feel free to make the necessary changes if you're looking at it now!

dannon committed:
Use stdioerrorlevel constants to indicate FATAL

dannon commented:
@caleb-easterly Sure, I'll do it now, thanks!

mvdbeek commented on lib/galaxy/tools/parser/interface.py:
```diff
@@ -5,6 +5,8 @@
 
 import six
 
+from galaxy.jobs.error_level import StdioErrorLevel
```
I was discussing that with @jmchilton yesterday, and we didn't want to import from galaxy.jobs to keep this separate. We could move the error_level code into tools though ?

dannon replied:
@mvdbeek Sure, that sounds good to me -- happy to move it wherever, I just want to use the constant.

jmchilton replied:
Pushed a fix for this, the refactoring I was hoping for actually eliminates some need for delayed imports and such so we closer to untangling circular imports in Galaxy I think. Thanks for working on this all!

jmchilton committed:
Refactor galaxy.jobs.error_level -> galaxy.tools.parser.error_level.

Needed after 46170e9bcdcc2eb104ebab72a7afc9b7238bc314 to keep galaxy-lib from importing things outside of galaxy-lib but clearly a good refactoring on its own since it eliminates mid-file imports that were there to prevent circular dependencies from loading Galaxy.

nsoranzo commented:
I've rebased and updated @jmchilton's last commit to also update the import in `test/unit/jobs/test_job_output_checker.py`
